gigaword-indexer/src/main/java/edu/isi/vista/gigawordIndexer/LTFDocuments.java

```diff
@@ -0,0 +1,117 @@
+package edu.isi.vista.gigawordIndexer;
+
+import com.google.common.collect.AbstractIterator;
+import com.google.common.collect.ImmutableList;
+import edu.isi.nlp.corpora.LctlText;
+import edu.isi.nlp.corpora.LtfDocument;
+import edu.isi.nlp.corpora.LtfReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.util.Enumeration;
+import java.util.Iterator;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LTFDocuments implements Iterable<Article>{
+
+  private static final Logger log = LoggerFactory.getLogger(LTFDocuments.class);
+
+  public static void main(String[] args) {
```

Comment:
If this is going to hold a reference to a zipfile, it needs to be `Closeable` so it can close the zipfile; then when you use it, you can put it in try-with-resources

Reply:
Removed. And made `LTFDocuments` `Closeable` and init it in a try-with-resource` block.